<?xml version="1.0"?>
<robot name="robot_mate">

  <!-- Definition du premier lien et joint -->
  <link name="Base">
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mate_description/meshes/Base.stl"/>
      </geometry>
      <material name="mat3">
        <color rgba="0.5 0.5 0.5 0.5"/> <!-- Grey color with 50% transparency -->
      </material>
    </visual>
  </link>

  <joint name="q_1" type="revolute">
    <parent link="Base"/>
    <child link="angle_motor"/>
    <origin xyz="0 0 0.1065" rpy="0 0 1.5708"/>  <!-- Rotation de 90° autour de Z -->
    <axis xyz="0 0 1"/>
    <limit effort="100" velocity="100" lower="-3.14159" upper="3.14159"/>
  </joint>

  <link name="angle_motor">
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 0 0.0" rpy="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin xyz="0 0 -0.11" rpy="0 0 -2.5"/>
      <geometry>
        <mesh filename="package://mate_description/meshes/angle_motor.stl"/>
      </geometry>
      <material name="mat4">
        <color rgba="0.5 0.5 0.5 0.5"/> <!-- Grey color with 50% transparency -->
      </material>
    </visual>
  </link>

  <link name="angle_motor_intermediary"/>

  <!-- Definition du repère intermediaire entre q1 et q2-->
  <joint name="q_2_intermediary" type="fixed">
    <parent link="angle_motor"/>
    <child link="angle_motor_intermediary"/>
    <origin xyz="0 0 0" rpy="-1.5708 0 0"/>  <!-- Rotation de -90° autour de X et de Z -->
  </joint>

  <!-- Definition de q2 -->
  <joint name="q_2" type="revolute">
    <parent link="angle_motor_intermediary"/>
    <child link="lower_arm"/>
    <origin xyz="0 0 -0.0855" rpy="0 0 -1.5708"/>  <!-- Rotation de -90° autour de X et de Z -->
    <axis xyz="0 0 1"/>
    <limit effort="100" velocity="100" lower="-1.57" upper="1.57"/>
  </joint>

  <link name="lower_arm">
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0.11 0 0" rpy="0 0 0"/>
      <inertia ixx="0.0001" ixy="0.0" ixz="0.0" iyy="0.0001" iyz="0.0" izz="0.0001"/>
    </inertial>
    <visual>
      <origin xyz="-0.11 -0.0 0.08" rpy="1.57 2.5 1.57"/>
      <geometry>
        <mesh filename="package://mate_description/meshes/lower_arm.stl"/>
      </geometry>
      <material name="mat1">
        <color rgba="0.5 0.5 0.5 0.5"/> <!-- Grey color with 50% transparency -->
      </material>
    </visual>
  </link>

  <!-- Definition du troisième link et joint q3 -->
  <joint name="q_3" type="revolute">
    <parent link="lower_arm"/>
    <child link="upper_arm"/>
    <origin xyz="0.220 0 0" rpy="0 0 1.5708"/>  <!-- Rotation de 90° autour de Z -->
    <axis xyz="0 0 1"/>
    <limit effort="100" velocity="100" lower="-1.6" upper="1.6"/>
  </joint>

  <link name="upper_arm">
    <inertial>
      <mass value="1.0"/>
      <origin xyz="0 -0.1 0" rpy="0 0 0"/>
      <inertia ixx="0.0001" ixy="0.0" ixz="0.0" iyy="0.0001" iyz="0.0" izz="0.0001"/>
    </inertial>
    <visual>
      <origin xyz="-0.18 0.23 0.08" rpy="1.57 2.55 0.12"/>
      <geometry>
        <mesh filename="package://mate_description/meshes/upper_arm.stl"/>
      </geometry>
      <material name="mat2">
        <color rgba="0.5 0.5 0.5 0.5"/> <!-- Grey color with 50% transparency -->
      </material>
    </visual>
  </link>

  <!-- Definition de la frame du tool (fixed joint) -->
  <joint name="q_4_tool_intermediary" type="fixed">
    <parent link="upper_arm"/>
    <child link="tool_intermediary"/>
    <origin xyz="0 0 0" rpy="1.5708 0 0"/>  <!-- Rotation de 90° autour de X -->
  </joint>

  <link name="tool_intermediary"/>

  <joint name="q_4_tool" type="fixed">
    <parent link="tool_intermediary"/>
    <child link="tool"/>
    <origin xyz="0 0 0.200" rpy="0 0 0"/>
  </joint>

  <link name="tool">
    <inertial>
      <mass value="0.0"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>

  <!-- Definition du cinquième lien et joint -->
  <joint name="q_5" type="fixed">
    <parent link="tool"/>
    <child link="end_effector"/>
    <origin xyz="0 0 0.0615" rpy="0 0 -1.5708"/>  <!-- Rotation de -90° autour de Z -->
  </joint>

  <link name="end_effector">
    <inertial>
      <mass value="0.0"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>

</robot>
